File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/PongDemoXml.kt
//
// PongDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class PongDemoXml {

    lateinit var customView: com.lightningkite.butterfly.views.widget.CustomView
    lateinit var customViewDelegate: com.lightningkite.butterflytemplate.vg.PongDelegate
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.pong_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        customView = view.findViewById<com.lightningkite.butterfly.views.widget.CustomView>(R.id.customView)
        customViewDelegate = view.findViewById<CustomView>(R.id.customView).delegate as com.lightningkite.butterflytemplate.vg.PongDelegate
        
        
        return view
    }
}

File: web/src/layout/PongDemoXml.ts
//
// PongDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import { PongDelegate } from '../vg/PongDemoVG'
import htmlForDefault from './pong_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.PongDemoXml
export class PongDemoXml {
    xmlRoot!: HTMLElement
    customView!: HTMLCanvasElement
    customViewDelegate!: PongDelegate
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.customView = getViewById<HTMLCanvasElement>(view, "customView")
        if(this.customView){ this.customViewDelegate = new PongDelegate(); customViewSetDelegate(this.customView, this.customViewDelegate); }
        return view
    }
}
